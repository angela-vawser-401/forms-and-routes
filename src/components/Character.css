.Character {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 80%;
  max-width: 400px;
  margin: auto;

  img {
    max-height: 300px;
    width: auto;
    border: 3px solid black;
    margin-top: 40px;
  }

  h1 {
    text-align: center;
    color: black;
    font-weight: bold;
    font-size: 1.75em;
    text-transform: uppercase;
    font-weight: 700;
    font-family: sans-serif;
    margin-top: 1em;
    margin-bottom: .5em;
  }

  button {
    text-transform: uppercase;
    padding: 5px;
    background-color: black;
    font-weight: 700;
    color: white;
    font-size: 16px;
    display: block;
    width: 250px;
  }
}
