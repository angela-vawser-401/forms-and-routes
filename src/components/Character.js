import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ character }) => {
  return (
    <div>
      <img src={character.photoUrl} />
      <h2>{character.name}</h2>
    </div>
  );
};

Character.propTypes = {
  character: PropTypes.object.isRequired
};

export default Character;
