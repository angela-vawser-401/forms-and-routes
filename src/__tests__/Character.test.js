import React from 'react';
import { shallow } from 'enzyme';
import Character from '../components/Character';

describe('Character component', () => {
  it('renders a character', () => {
    const handleClick = () => {
      // eslint-disable-next-line no-console
      console.log('peekaboo');
    };
    const wrapper = shallow(
      <Character 
        name="Title"
        photo="Image"
        handleClick={handleClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  
});
