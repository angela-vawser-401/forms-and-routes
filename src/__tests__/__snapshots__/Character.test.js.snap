// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<div
  className="Character"
>
  <img
    src="Image"
  />
  <h1>
    Title
  </h1>
  <button
    onClick={[Function]}
  >
    Generate Character
  </button>
</div>
`;
